packages/epics/src/membership/components/members/member-detail.tsx

```diff
@@ -0,0 +1,163 @@
+import { MemberHead } from './member-head';
+import { Skeleton, Button, Separator, FilterMenu } from '@hypha-platform/ui';
+import { RxCross1 } from 'react-icons/rx';
+import { Image } from '@hypha-platform/ui';
+import { SectionLoadMore } from '@hypha-platform/ui/server';
+import { AgreementItem } from '@hypha-platform/graphql/rsc';
+import { useMemberDetails } from '../../hooks/use-member-details';
+import Link from 'next/link';
+import { AgreementsList } from '../../../agreements/components/agreements-list';
+
+type SpaceType = {
+  name?: string;
+  logo?: string;
+};
+
+type MemberType = {
+  avatar?: string;
+  name?: string;
+  surname?: string;
+  nickname?: string;
+  commitment?: number;
+  status?: string;
+  about?: string;
+  spaces?: SpaceType[];
+  agreements?: AgreementItem[];
+};
+
+export type MemberDetailProps = {
+  closeUrl: string;
+  member: MemberType;
+  agreements: AgreementItem[];
+  isLoading: boolean;
+};
+
+export const MemberDetail = ({
+  isLoading,
+  closeUrl,
+  member,
+}: MemberDetailProps) => {
+  const {
+    loadMore,
+    pagination,
+    paginatedAgreements,
+    activeFilter,
+    setActiveFilter,
+  } = useMemberDetails(member.agreements ?? []);
```

Comment:
We should discuss this architecture. From my point of view, that is a weird/unclear data handling.